display_blocks.js

```diff
@@ -161,10 +157,116 @@ function parseObject(obj) {
+			colour = "LIGHTBLUE";
+			break;
+		case "PAUSED":
+			colour = "RED";
+			break;
+		case "WAITING" || "VETOING":
+			colour = "GOLDENROD";
+			break;
+		case "ENDING" || "ABORTING":
+			colour = "BLUE";
+			break;
+		case "PAUSING":
+			colour = "DARK_RED";
+			break;
+	};
+	document.getElementById("inst_name").style = "padding: 10px; background-color:" +colour+"; border: black 2px solid";
+	var title = document.createElement("h3"); 
+	title.innerHTML = instrument.toUpperCase()+" is "+runStatus;
+	var blockListClass = document.createAttribute("class");
+	blockListClass.value = "text-center";
+	title.setAttributeNode(blockListClass);
+	document.getElementById("inst_name").appendChild(title);
+	
+	var runNum = document.createElement("h3"); 
+	runNum.innerHTML = "Run: "+obj["inst_pvs"]["RUNNUMBER"]["value"];
+	document.getElementById("inst_name").appendChild(runNum);
+	
+	var titleName = document.createElement("h5"); 
+	insidePart ="Title: "+obj["inst_pvs"]["TITLE"]["value"];
+	titleName.innerHTML = insidePart + "\xa0".repeat(30);
+	document.getElementById("next_part").appendChild(titleName);
+	
+	var userName = document.createElement("h5");
+	insidePart = "Users: "+obj["inst_pvs"]["_USERNAME"]["value"];
+	userName.innerHTML = insidePart + "\xa0".repeat(30);
+	document.getElementById("next_part").appendChild(userName);
+	
+	newPartOfTable();
+	var goodRawFrames = document.createElement("h5"); 
+	insidePart ="Good / Raw Frames: "+obj["inst_pvs"]["GOODFRAMES"]["value"]+"/"+obj["inst_pvs"]["RAWFRAMES"]["value"];
+	titleName.innerHTML = insidePart + "\xa0".repeat(30);
+	document.getElementById("next_part").appendChild(titleName);
+
+	var currentTotal = document.createElement("h5");
+	insidePart ="Current / Total: "+obj["inst_pvs"]["BEAMCURRENT"]["value"]+"/"+obj["inst_pvs"]["TOTALUAMPS"]["value"];
+	currentTotal.innerHTML = insidePart + "\xa0".repeat(30);
+	document.getElementById("next_part").appendChild(currentTotal);
+	
+	var monitorCount = document.createElement("h5");
+	insidePart ="Monitor Counts: "+obj["inst_pvs"]["MONITORCOUNTS"]["value"];
+	monitorCount.innerHTML = insidePart + "\xa0".repeat(30);
+	document.getElementById("next_part").appendChild(monitorCount);
+	
+	newPartOfTable();
+	var instTime = document.createElement("h5"); 
+	insidePart ="Inst. Time: "+obj["inst_pvs"]["STARTTIME"]["value"];
+	instTime.innerHTML = insidePart + "\xa0".repeat(30);
+	document.getElementById("next_part").appendChild(instTime);
+
+	var runTime = document.createElement("h5");
+	insidePart ="Run Time: "+obj["inst_pvs"]["RUNDURATION_PD"]["value"];
+	runTime.innerHTML = insidePart + "\xa0".repeat(30);
+	document.getElementById("next_part").appendChild(runTime);
+	
+	var currentTotalPeriods = document.createElement("h5");
+	insidePart ="Period: "+obj["inst_pvs"]["PERIOD"]["value"]+"/"+obj["inst_pvs"]["NUMPERIODS"]["value"];
+	currentTotalPeriods.innerHTML = insidePart + "\xa0".repeat(30);
+	document.getElementById("next_part").appendChild(currentTotalPeriods);
```

Comment:
A lot of these bits look pretty close to identical. See if they can be refactored to use a common method.